tests/acceptance/features/bootstrap/GraphContext.php

```diff
@@ -1240,4 +1240,191 @@ public function checkUserInformation(array $expectedValue, array  $actualValue):
+		$this->featureContext->setResponse($response);
+	}
+
+	/**
+	 *
+	 * @When /^the user "([^"]*)" gets (his|her) drive information using Graph API$/
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 */
+	public function userTriesToGetOwnDriveInformation(string $user) {
+		$response = $this->retrieveUserInformationAlongWithDriveUsingGraphApi($user);
+		$this->featureContext->setResponse($response);
+	}
+
+	/**
+	 * @param array $driveInformation
+	 *
+	 * @return string
+	 */
+	public static function getSpaceIdFromActualDriveinformation(array $driveInformation): string {
+		return $driveInformation['id'];
+	}
+
+	/**
+	 * check if single drive information is correct
+	 *
+	 * @param array $expectedDriveInformation
+	 * @param array $actualDriveInformation
+	 *
+	 * @return void
+	 */
+	public function checkUserDriveInformation(array $expectedDriveInformation, array  $actualDriveInformation):void {
+		foreach (array_keys($expectedDriveInformation) as $keyName) {
+			// break the segment with @@@  to find the actual value from the drive infromation from response
+			$separatedKey = explode("@@@", $keyName);
+			// this stores the actual value of each key from drive information response used for assertion
+			$actualKeyValue = $actualDriveInformation;
+
+			foreach ($separatedKey as $key) {
+				$actualKeyValue = $actualKeyValue[$key];
+			}
+
+			switch ($expectedDriveInformation[$keyName]) {
+				case '%user_id%':
+					Assert::assertTrue(GraphHelper::isUUIDv4($actualKeyValue), __METHOD__ . ' Expected user_id to have UUIDv4 pattern but found: ' . $actualKeyValue);
+					break;
+				case '%space_id%':
+					Assert::assertTrue(GraphHelper::isSpaceId($actualKeyValue), __METHOD__ . ' Expected space_id to have a UUIDv4:UUIDv4 pattern but found: ' . $actualKeyValue);
+					break;
+				default:
+					$expectedDriveInformation[$keyName] = $this->featureContext->substituteInLineCodes(
+						$expectedDriveInformation[$keyName],
+						$this->featureContext->getCurrentUser(),
+						[],
+						[
+							[
+								// the actual space_id is substituted from the actual drive information rather than making an API request and substituting
+								"code" => "%space_id%",
+								"function" =>
+									[$this, "getSpaceIdFromActualDriveinformation"],
+								"parameter" => [$actualDriveInformation]
+							],
+						]
+					);
+					Assert::assertEquals($expectedDriveInformation[$keyName], $actualKeyValue);
+			}
+		}
+	}
+
+	/**
+	 * @param TableNode $table
+	 *
+	 * @Then the user retrieve API response should contain the following drive information:
+	 *
+	 * @return void
+	 */
+	public function theResponseShouldContainTheFollowingDriveInformation(TableNode $table): void {
+		$expectedDriveInformation = $table->getRowsHash();
+		// array of user drive information (Personal Drive Information Only)
+		$actualDriveInformation = $this->featureContext->getJsonDecodedResponse($this->featureContext->getResponse())['drive'];
```

Comment:
check first the `$this->featureContext->getJsonDecodedResponse($this->featureContext->getResponse())` statement is array and has `drive` key in it.. will be more easier for debugging in case of malformed responses.